### ISSUE TITLE
Incorporate FieldRange within filters

### ISSUE THREAD

**Issue by friek108** [kartik-v/yii2-grid#383]
Wondering if there is any chance to enhance the filters used to include FieldRange?
https://github.com/andrewblake1/yii2-field-range/commit/fea414ead4ce72557b5148224e429e9a18cc2ba6

Maybe a better implementation would be to have one 'field' but add the ability to put a mask on it separated by a dash (maybe / maybe not).

**Comment by andrewblake1** [comment]
Note that the code in the link above is outdated (simplified with just the css below and setting the input placeholders ).

IMHO having the option for stacked inputs is very useful for field range - in gridview particularly. 1 input wide should be enough for the column. The second input undesirably (particularly on mobiles) expands the column.

```
.kv-grid-table  thead .input-daterange input  {
    text-align: left;
}
.kv-grid-table  thead .input-group .form-control:first-child {
    border-top-right-radius: 4px !important;
    border-top-left-radius: 4px !important;
    border-bottom-right-radius: 0 !important;
    border-bottom-left-radius: 0 !important;
    border-bottom: none;
}
.kv-grid-table  thead .input-group .form-control:not(:first-child) {
    border-top-right-radius: 0 !important;
    border-top-left-radius: 0 !important;
    border-bottom-right-radius: 4px !important;
    border-bottom-left-radius: 4px !important;
}
.kv-grid-table  thead .form-group {
    margin-bottom: 0;
}
.kv-grid-table  thead .help-block {
    margin-top: 0;
    margin-bottom: 0;
}
.kv-grid-table  thead .form-control.kv-field-range.input-group {
    padding: 0;
    border: none;
}
.kv-grid-table  thead span.input-group-addon.kv-field-separator {
    display: none;
}
.kv-grid-table  thead .form-group label {
    display: none;
}
```

**Comment by kartik-v** [comment]
As a thought, I had intended the following widgets can be used for range for specific data types. 

- **Date Time Range**: [kartik-v/yii2-date-range](http://demos.krajee.com/date-range)
- **Number Range**: [kartik-v/yii2-slider](http://demos.krajee.com/slider)

For string based fields a range does not make a sense... but I feel you can use
- **String**: [kartik-v/yii2-widget-select2](http://demos.krajee.com/widget-details/select2) with multiple option 

For styling issues the field range widget may not fit in grid cells for all scenarios. But you can still use the FieldRange widget as is by using 2 different attributes on your Grid searchModel that renders the data provider to filter the query.

**Comment by conick** [comment]
@kartik-v it is not the way when I want to use [min - max] (money/number) filter like as DateRange filter. Could you suggest any solution for this case? I guess Slider is for one exact value.

**Comment by kartik-v** [comment]
No Slider can be configured as a range... with a min max value... refer yii2-slider docs and demos and [check the 3rd example](http://demos.krajee.com/slider#usage)

**Comment by conick** [comment]
@kartik-v Unfortunately slider fires change event and submits grid filters form on headway on every degree even my mouse still in clicked state. Is it bug? I think that it's normal behaviour is when it fires onmouseup. With current behaviour it cannot be used with pjax. The [similiar problem](https://github.com/kartik-v/yii2-widget-select2/issues/134) with Select2 - it fires twice on clearing input.
